### ISSUE TITLE
Modify search engine to relevance ranking

### ISSUE THREAD

**Issue by tiigit** [rrimpila/tiramisu#21]
Modify the code to use relevance ranking based on tf-idf and cosine similarity instead of Boolean operators.

**Comment by rrimpila** [comment]
I moved the boolean search functions under their own method for (potentially) easier use if/when we enable choosing between them. 
ranking_search method has the most new code. Please test. 
The code could use some refactoring as well, this is just a quick and dirty implementation.

**Comment by tiigit** [comment]
I tested the ranking_search function and handled some errors that occurred (1. search queries that were only special characters such as "!?" because they couldn't be handled by the tfidvectorizer, 2. search words that were combinations of letters and numbers such as "cat1", because also those weren't handled by tdifv). The ranking_search does not find one-letter words so at least that issue is still to be fixed, if either of you wants to work on that? :) @rrimpila @ciccmin 

EDIT: For unknown words: right now the IndexError clause that I added catches those and prints out an error message, but if we want to handle those differently, we can work on that as well.
